contracts/upgradeable_contracts/arbitrary_message/MessageDelivery.sol

```diff
@@ -0,0 +1,43 @@
+pragma solidity 0.4.24;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract MessageDelivery {
+    using SafeMath for uint256;
+
+    function requireToPassMessage(address _contract, bytes _data, uint256 _gas) public {
+        require(isMessageDeliverySubsidizedMode());
+        require(_gas >= getMinimumGasUsage(_data) && _gas <= getMaxGasPerTx());
```

Comment:
can we use here `maxGasPerTx`?

Reply:
Fixed here https://github.com/poanetwork/poa-bridge-contracts/pull/77/commits/e1eb585dd20c84e8686717424225c25bfc1ea3d7